### PR TITLE
do not enrich html formatters if is not jupyter 

### PR DESCRIPTION
**Pull request by colombod**
When starting with http api enabled the formatting of js and html is enriched only during the execution  of juptyer command

### CODE REVIEW

File: src/dotnet-interactive.Tests/FormatterConfigurationTests.cs [hunk 1]
```diff
@@ -81,11 +81,12 @@ public void ScriptContent_type_is_formatted()
         }
 
         [Fact]
-        public void ScriptContent_type_is_wrapped_when_http_is_enabled()
+        public void ScriptContent_type_is_wrapped_when_http_is_enabled_and_flavor_is_jupyter()
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/FormatterConfigurationTests.cs:84]: Mmm. Flavor.

File: src/dotnet-interactive/CommandLine/BrowserFrontendEnvironment.cs [hunk 2]
```diff
@@ -8,5 +8,6 @@ namespace Microsoft.DotNet.Interactive.App.CommandLine
     public class BrowserFrontendEnvironment : FrontendEnvironment
     {
         public Uri ApiUri { get; set; }
+        public string Flavor { get; set; }
```
Comment (jonsequitur) [inline src/dotnet-interactive/CommandLine/BrowserFrontendEnvironment.cs:11]: This is pretty unclear in terms of how it should be understood or used.

File: src/dotnet-interactive.Tests/FormatterConfigurationTests.cs (hunk 1 above)
Comment (colombod) [reply src/dotnet-interactive.Tests/FormatterConfigurationTests.cs]: could have been flavour but it gone

File: src/dotnet-interactive/CommandLine/BrowserFrontendEnvironment.cs (hunk 2 above)
Comment (colombod) [reply src/dotnet-interactive/CommandLine/BrowserFrontendEnvironment.cs]: agreed

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 3]
```diff
@@ -163,6 +164,32 @@ public async Task http_command_uses_default_port()
...
+            using var scope = new AssertionScope();
+            _serviceCollection
+                .FirstOrDefault(s => s.ServiceType == typeof(BrowserFrontendEnvironment))
+                .Should()
+                .NotBeNull();
+
+        }
+
+        [Fact]
+        public async Task http_command_does_not_registers_JupyterFrontedEnvironment()
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:181]:
```suggestion
        public async Task http_command_does_not_register_JupyterFrontedEnvironment()
```

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 4]
```diff
@@ -163,6 +164,32 @@ public async Task http_command_uses_default_port()
...
+        {
+            await _parser.InvokeAsync($"http");
+
+            using var scope = new AssertionScope();
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:185]: The `AssertionScope` isn't needed.

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 5]
```diff
@@ -174,6 +201,24 @@ public void http_command__does_not_parse_http_port_range_option()
                  .Contain(errorMessage => errorMessage == "Unrecognized command or argument '--http-port-range'");
         }
 
+        [Fact]
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:204]: The more interesting thing to test here is what type do I get when I resolve the `FrontendEnvironment`?

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs [hunk 6]
```diff
@@ -250,8 +259,17 @@ Command HttpServer()
...
-                        RegisterKernelInServiceCollection(services, startupOptions, options.DefaultKernel);
-
+                        RegisterKernelInServiceCollection(
+                            services,
+                            startupOptions,
+                            options.DefaultKernel, serviceCollection =>
+                        {
+                            serviceCollection.AddSingleton(_ =>
+                            {
+                                var frontendEnvironment = new BrowserFrontendEnvironment();
```
Comment (jonsequitur) [inline src/dotnet-interactive/CommandLine/CommandLineParser.cs:269]: What happens when `FrontEndEnvironment` is resolved?

Comment (colombod) [reply src/dotnet-interactive/CommandLine/CommandLineParser.cs:269]: I still need to register for the discovery handler

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs (hunk 5 above)
Comment (colombod) [reply src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:204]: none as the parser fails and we do not invoke it

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs (hunk 4 above)
Comment (colombod) [reply src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:185]: removed

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs (hunk 6 above)
Comment (colombod) [reply src/dotnet-interactive/CommandLine/CommandLineParser.cs:268]: all subtype of FrontendEnvironment are also register with as base type and resolved as base type

**Pull request merged**